apps/studio/components/interfaces/Database/Backups/RestoreToNewProject/CreateNewProjectDialog.tsx

```diff
@@ -0,0 +1,212 @@
+import { toast } from 'sonner'
+import { z } from 'zod'
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import PasswordStrengthBar from 'components/ui/PasswordStrengthBar'
+import { useProjectCloneMutation } from 'data/projects/clone-mutation'
+import { useCloneBackupsQuery } from 'data/projects/clone-query'
+import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+import { passwordStrength } from 'lib/helpers'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogTitle,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input,
+  Input_Shadcn_,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import { AdditionalMonthlySpend } from './AdditionalMonthlySpend'
+
+interface CreateNewProjectDialogProps {
+  open: boolean
+  selectedBackupId: number | null
+  recoveryTimeTarget: number | null
+  onOpenChange: (value: boolean) => void
+  onCloneSuccess: () => void
+}
+
+export const CreateNewProjectDialog = ({
+  open,
+  selectedBackupId,
+  recoveryTimeTarget,
+  onOpenChange,
+  onCloneSuccess,
+}: CreateNewProjectDialogProps) => {
+  const { project } = useProjectContext()
+  const organization = useSelectedOrganization()
+
+  const [passwordStrengthScore, setPasswordStrengthScore] = useState(0)
+  const [passwordStrengthMessage, setPasswordStrengthMessage] = useState('')
+
+  const FormSchema = z.object({
+    name: z.string().min(1),
+    password: z.string().min(1),
+  })
+
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      name: '',
+      password: '',
+    },
+  })
+
+  const { data: subscription } = useOrgSubscriptionQuery({ orgSlug: organization?.slug })
+  const isFreePlan = subscription?.plan?.id === 'free'
+
+  const { data: cloneBackups } = useCloneBackupsQuery(
+    { projectRef: project?.ref },
+    { enabled: !isFreePlan }
+  )
+  const hasPITREnabled = cloneBackups?.pitr_enabled
+
+  const { mutate: triggerClone, isLoading: cloneMutationLoading } = useProjectCloneMutation({
+    onError: (error) => {
+      console.error('error', error)
+      toast.error('Failed to restore to new project')
+    },
+    onSuccess: () => {
+      toast.success('Restoration process started')
+      onCloneSuccess()
+      // refetchCloneStatus()
+      // setShowNewProjectDialog(false)
+    },
+  })
+
+  const delayedCheckPasswordStrength = useRef(
+    debounce((value: string) => checkPasswordStrength(value), 300)
+  ).current
+
+  async function checkPasswordStrength(value: string) {
+    const { message, strength } = await passwordStrength(value)
+    setPasswordStrengthScore(strength)
+    setPasswordStrengthMessage(message)
+  }
+
+  function generateStrongPassword() {
```

Comment:
doesn't have to be now, but shouldn't all this be in some sort of project.utils

we have dup code here, in vercel integration pages, and project create

Reply:
yeah you're right here - should be de-duping! i'll look into in this PR actually